CRAYSAT-1624: Document errors and warnings in bos-operations stage (#2838)
Cray-HPE/docs-csm: operations/power_management/Shut_Down_and_Power_Off_Compute_and_User_Access_Nodes.md

## Prerequisites

The `cray` and `sat` commands must be initialized and authenticated with valid credentials for Keycloak. If these have not been prepared, then see

> [Configure the Cray Command Line Interface (`cray` CLI)](../configure_cray_cli.md) and refer to the "SAT Authentication" section of the HPE Cray EX
> System Admin Toolkit (SAT) product stream documentation (S-8031) for instructions on how to acquire a SAT authentication token.

Reviewer: ```suggestion
[Configure the Cray Command Line Interface (`cray` CLI)](../configure_cray_cli.md) and refer to the "SAT Authentication" section of the *HPE Cray EX
System Admin Toolkit (SAT) (S-8031)* product stream documentation for instructions on how to acquire a SAT authentication token.
```

   To monitor the progress of this job, run the following command in a separate window:
       'kubectl -n services logs -c boa -f --selector job-name=boa-a1a697fc-e040-4707-8a44-a6aef9e4d6ea'

   Waiting for BOA k8s job with id boa-79584ffe-104c-4766-b584-06c5a3a60996 to complete. Session template: cos-2.0.0.
   To monitor the progress of this job, run the following command in a separate window:
       'kubectl -n services logs -c boa -f --selector job-name=boa-79584ffe-104c-4766-b584-06c5a3a60996'

   [...]

   All BOS sessions completed.
   ```

   **Note:** In certain cases, the command may display an error similar to the following:

   ```text
   ERROR: Failed to get state of nodes in session template 'UAN_SESSION_TEMPLATE': Failed to get state of nodes with role=['Application', 'Application_UAN'] for boot set 'BOOT_SET' of session template 'UAN_SESSION_TEMPLATE': GET request to URL 'https://api-gw-service-nmn.local/apis/smd/hsm/v2/State/Components' failed with status code 400: Bad Request. Bad Request Detail: bad query param: Argument was not a valid HMS Role
   ```

>    This error is a non-fatal error and does not affect the `bos-operations` stage of `sat bootsys`.

Reviewer: ```suggestion
   This is a non-fatal error and does not affect the `bos-operations` stage of `sat bootsys`.
```

   **Note:** In certain cases, the command may fail before reaching the displayed timeout

>    and will show warnings similar to the following:

Reviewer: ```suggestion
   and show warnings similar to the following:
```

   ```text
   WARNING: The 'kubectl wait' command failed instead of timing out. stderr: error: condition not met for jobs/boa-79584ffe-104c-4766- b584-06c5a3a60996
   ```

>    The existence of these warnings does not prevent the BOS operation from proceeding, however it
>    may result in the `bos-operations` stage of the `sat bootsys` command exiting before the BOS
>    operation is complete. In this case it is especially important to view the logs to monitor the
>    shutdown, and to verify that the nodes reached the expected power state, both of which are shown
>    in the remaining steps.

Reviewer: ```suggestion
    The BOS operation can still proceed even with these warnings. However, the warnings
    may result in the `bos-operations` stage of the `sat bootsys` command exiting before the BOS
    operation is complete. Because of this, it is important to view the logs in order to monitor the
    boot and to verify that the nodes reached the expected state. Both of these recommendations are shown
    in the remaining steps.
```

> 1. Use the Job ID strings \(for example, the `cos-2.0.0` session, `boa-79584ffe-104c-4766- b584-06c5a3a60996`\) from the previous command to monitor the progress of the shutdown of the compute nodes.

Reviewer: ```suggestion
1. Use the Job ID strings from the previous command to monitor the shutdown progress of the compute nodes.

   For example, use the `cos-2.0.0` session `boa-79584ffe-104c-4766- b584-06c5a3a60996`.
```

   2020-08-21 17:27:02,395 - INFO    - cray.boa.connection - Reattempting GET request for 'http://cray-cfs-api/apis/cfs/sessions'
   2020-08-21 17:27:02,437 - INFO    - cray.boa.connection - Reattempting GET request for 'http://cray-cfs-api/apis/cfs/sessions'
   2020-08-21 17:27:02,519 - INFO    - cray.boa.connection - Reattempting GET request for 'http://cray-cfs-api/apis/cfs/sessions'
   2020-08-21 17:27:02,681 - INFO    - cray.boa.connection - Reattempting GET request for 'http://cray-cfs-api/apis/cfs/sessions'
   2020-08-21 17:27:03,003 - INFO    - cray.boa.connection - Reattempting GET request for 'http://cray-cfs-api/apis/cfs/sessions'
   2020-08-21 17:27:03,645 - INFO    - cray.boa.connection - Reattempting GET request for 'http://cray-cfs-api/apis/cfs/sessions'

   ```

   The BOS shutdown session may or may not power off compute nodes depending on the session template being used.

1. In another shell window, use a similar command to monitor the UAN session.

   ```bash
   kubectl -n services logs -c boa -f --selector job-name=boa-a1a697fc-e040-4707-8a44-a6aef9e4d6ea
   ```

> 1. Check the status of UAN and compute nodes to verify they are `Off`.

Reviewer: ```suggestion
1. Check the status of both UAN and compute nodes to verify that they are `Off`.
```

>    There may be delay in nodes reaching the `Off` state in the hardware state manager \(HSM\).

Reviewer: ```suggestion
   There may be a delay in nodes reaching the `Off` state in the hardware state manager \(HSM\).
```
